src/libs/actions/IOU.js

```diff
@@ -75,11 +75,29 @@ Onyx.connect({
     },
 });
 
+let didInitCurrency = false;
+Onyx.connect({
+    key: ONYXKEYS.IOU,
+    callback: (val) => {
+        didInitCurrency = lodashGet(val, 'didInitCurrency');
+    },
+});
+
+let shouldResetIOUAfterLogin = true;
 let currentUserPersonalDetails = {};
 Onyx.connect({
     key: ONYXKEYS.PERSONAL_DETAILS_LIST,
     callback: (val) => {
         currentUserPersonalDetails = lodashGet(val, userAccountID, {});
+        if (!val || !shouldResetIOUAfterLogin || didInitCurrency) {
+            return;
+        }
+        // eslint-disable-next-line no-use-before-define
+        resetMoneyRequestInfo();
+        shouldResetIOUAfterLogin = false;
```

Comment:
I am concerned that this is a hack that does not address the root cause of the problem. I think it is a mistake that the IOU in Onyx is the thing in charge of initializing currency.

I suggest reverting this PR since it has already been linked to a regression: https://github.com/Expensify/App/issues/28280 and go back to the drawing board for a better solution.

Alternatively, you can wait for me to make progress and resolve https://github.com/Expensify/App/issues/26538 which is an attempt to clean up a lot of the messing routing logic that is probably closer to the root of the problems.